src/main/java/org/craftercms/deployer/impl/lifecycle/aws/DuplicateS3LifecycleHook.java

```diff
@@ -0,0 +1,158 @@
+    protected static final String CONFIG_KEY_URL = "url";
+    protected static final String DELIMITER = "/";
+    private final String siteName;
+    private final String sourceSiteName;
+    private final ProcessedCommitsStore processedCommitsStore;
+    private final TargetService targetService;
+    private final String blobExtension;
+    protected final ThreadPoolTaskExecutor threadPoolTaskExecutor;
+
+    private boolean ignoreBlobs;
+    private AwsClientBuilderConfigurer<AmazonS3ClientBuilder> builderConfigurer;
+    private AmazonS3URI s3Url;
+    private AmazonS3URI srcS3Url;
+    private String srcLocalRepoPath;
+
+    @ConstructorProperties({"siteName", "sourceSiteName", "processedCommitsStore",
+            "targetService", "blobExtension", "threadPoolTaskExecutor"})
+    public DuplicateS3LifecycleHook(String siteName, String sourceSiteName, ProcessedCommitsStore processedCommitsStore,
+                                    TargetService targetService, String blobExtension, ThreadPoolTaskExecutor threadPoolTaskExecutor) {
+        this.siteName = siteName;
+        this.sourceSiteName = sourceSiteName;
+        this.processedCommitsStore = processedCommitsStore;
+        this.targetService = targetService;
+        this.blobExtension = blobExtension;
+        this.threadPoolTaskExecutor = threadPoolTaskExecutor;
+    }
+
+    @Override
+    protected void doInit(Configuration config) throws ConfigurationException, DeployerException {
+        builderConfigurer = new AwsS3ClientBuilderConfigurer(config);
+        s3Url = new AmazonS3URI(appendIfMissing(getRequiredStringProperty(config, CONFIG_KEY_URL), DELIMITER));
+
+        ignoreBlobs = getBooleanProperty(config, CONFIG_KEY_IGNORE_BLOBS, true);
+
+        Configuration srcTargetConfig = config.subset(CONFIG_KEY_SOURCE_CONFIG);
+        srcS3Url = new AmazonS3URI(appendIfMissing(getRequiredStringProperty(srcTargetConfig, CONFIG_KEY_URL), DELIMITER));
+        srcLocalRepoPath = getRequiredStringProperty(srcTargetConfig, CONFIG_KEY_LOCAL_REPO_URL);
+
+    }
+
+    protected AmazonS3 buildClient(AwsClientBuilderConfigurer<AmazonS3ClientBuilder> builderConfigurer) {
+        AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
+        builderConfigurer.configureClientBuilder(builder);
+        return builder.build();
+    }
+
+    /**
+     * Get the list of paths to copy between buckets.
+     * This method will list all files in the source bucket (filtering out .blob files if ignoreBlobs is true).
+     *
+     * @return List of paths to copy between buckets
+     * @throws DeployerException If an error occurs while retrieving the list of paths
+     */
+    private List<String> getItemPathList(Target target) throws DeployerException {
+        Target srcTarget = targetService.getTarget(target.getEnv(), sourceSiteName);
+
+        List<String> paths = new LinkedList<>();
+        try (Git git = GitUtils.openRepository(new File(srcLocalRepoPath))) {
```

Comment:
JGit tree walks are really expensive. Can we just get a list with all the files of the source S3 bucket path?

Reply:
And if that's not possible, let's just read directly from the filesystem

Reply:
@sumerjabri @avasquez614 
Updated so it gets the list from the filesystem. Listing from the FS was chosen instead of listing the S3 bucket since some contents will be owned by Studio. Also, repo (either via GIT or FS) should a better source of truth than S3.